websites/M/MangaFire/presence.ts

```diff
@@ -0,0 +1,119 @@
+const presence = new Presence({ clientId: "1080885668248559648" }),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+enum Assets {
+	Logo = "https://i.imgur.com/KXGOlbM.png",
+	Searching = "https://i.imgur.com/OIgfjTG.png",
+	Reading = "https://i.imgur.com/53N4eY6.png",
+}
+
+presence.on("UpdateData", async () => {
+	const { pathname } = document.location,
+		pathArr = pathname.split("/"),
+		{ details, smallImageKey, largeImageKey, state } = getPageData(
+			pathArr[1],
+			pathArr[2],
+			document.querySelector(".name")?.textContent
+		),
+		presenceData: PresenceData = {
+			largeImageKey: largeImageKey || Assets.Logo,
+			startTimestamp: browsingTimestamp,
+			details,
+		},
+		[showDetail] = await Promise.all([presence.getSetting<boolean>("details")]);
```

Comment:
Promise all ain't needed for 1 promise

Reply:
Ah, my mistake. I had another setting but deleted it and forgot to change it. thx